openreview/profile/process/request_remove_name_decision_process.py

```diff
@@ -0,0 +1,222 @@
+def process(client, note, invitation):
+
+    SUPPORT_USER_ID = ''
+    AUTHOR_RENAME_INVITATION_ID = ''
+    request_note = client.get_note(note.referent)
+    usernames = request_note.content['usernames']
+    profile = client.get_profile(usernames[0])
+    preferred_name = profile.get_preferred_name(pretty=True)
+    
+    if 'Rejected' == request_note.content['status']:
+        client.post_message(subject='Profile name removal request has been rejected', 
+        recipients=[profile.id], 
+        message=f'''Hi {{{{fullname}}}},
+
+We have received your request to remove the name "{request_note.content['name']}" from your profile: https://openreview.net/profile?id={profile.id}.
+
+We can not remove the name from the profile for the following reason:
+
+{request_note.content['support_comment']}
+
+Regards,
+
+The OpenReview Team.
+''')
+        return       
+    
+    for username in usernames:
+        print('Check if we need to rename the profile')
+        if username == profile.id:
+            profile = client.rename_profile(profile.id, profile.get_preferred_name())
+        
+        print('Replace all the publications that contain the name to remove')
+        publications = client.get_all_notes(content={ 'authorids': username})
+        for publication in publications:
+            authors = []
+            authorids = []
+            needs_change = False
+            for index, author in enumerate(publication.content.get('authorids')):
+                if username == author:
+                    authors.append(preferred_name)
+                    authorids.append(profile.id)
+                    needs_change = True
+                else:
+                    if publication.content.get('authors'):
+                        authors.append(publication.content['authors'][index])
+                    authorids.append(publication.content['authorids'][index])
+            if needs_change:
+                content = {
+                    'authors': authors,
+                    'authorids': authorids
+                }
+                if '_bibtex' in publication.content:
+                    content['_bibtex'] = publication.content['_bibtex']['value'].replace(openreview.tools.pretty_id(username), preferred_name)                
```

Comment:
This line threw an error for me when I tested it on real profile data:
TypeError: string indices must be integers

Reply:
sorry about this, I tested with the API 2 but API 1. I pushed the fix, please try again.

Reply:
That worked!